### PR TITLE
[Serve] Expose ApplicationStatus class to API docs (#34966)

### LINKED ISSUE #34966: [Serve] Application status info is missing in the api reference
### Description

Screen Shot 2023-05-02 at 11 21 42 AM

### Link

_No response_

### PR DESCRIPTION
**Pull request by akyang-anyscale**
## Why are these changes needed?

This PR moves the ApplicationStatus class out of the _private directory, allowing it to be included in the API docs.

## Related issue number

Closes https://github.com/ray-project/ray/issues/34966

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — APPROVED** [review]
A journey of a thousand miles...

**Comment by edoakes** [comment]
Looks like a couple annotations are still missing: https://buildkite.com/ray-project/microcheck/builds/6362#0192bb23-7c79-4009-898b-cd8969cf276f/184-476

**Comment by akyang-anyscale** [comment]
Moved `ApplicationStatusInfo` and `StatusOverview` back into `_private/` since these are only used in `_private/`. They now live in `application_state.py`.

File: python/ray/serve/schema.py [hunk 1]
```diff
@@ -824,10 +823,22 @@ class DeploymentStatusOverview:
...
+    RUNNING = "RUNNING"
+    UNHEALTHY = "UNHEALTHY"
+    DELETING = "DELETING"
+
+
 @PublicAPI(stability="alpha")
 @dataclass
 class ApplicationStatusOverview:
-    """Describes the status of an application and all its deployments.
+    """Describes the satus of an application and all its deployments.
```
Comment (zcin) [inline python/ray/serve/schema.py:841]:
```suggestion
    """Describes the status of an application and all its deployments.
```

**Review by zcin — APPROVED** [review]
Nice!

File: python/ray/serve/schema.py (hunk 1 above)
Comment (akyang-anyscale) [reply python/ray/serve/schema.py:841]: nice catch

**Pull request merged**